### PR TITLE
Family onboarding

### PR DESCRIPTION
**Pull request by chennara**
Depends on: https://github.com/cultuurnet/uitpas-mobile-app/pull/285

### CODE REVIEW

File: src/onboarding/family/FamilyOnboarding.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  { text: 'ONBOARDING.FAMILY.TEXT_4' },
+];
+
+export const FamilyOnboarding = () => {
+  const { isLoading, isFetchedAfterMount, data: hasFamilyMembers } = useHasFamilyMembers();
+  const { dismissFamilyOnboarding } = useOnboarding();
+
+  useEffect(() => {
+    if (isFetchedAfterMount) {
+      storage.set(StorageKey.HasSeenFamilyOnboarding, true);
```
[the remark this answers is not in the document]
Comment (chennara) [reply src/onboarding/family/FamilyOnboarding.tsx]: Nice catch, fixed.

File: src/onboarding/family/_queries/useHasFamilyMembers.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { useState } from 'react';
+
+import { usePubliqApi } from '../../../_hooks/usePubliqApi';
+import { TFamilyMember } from '../../../profile/_models';
+import { useGetMe } from '../../../profile/_queries/useGetMe';
+
+export const useHasFamilyMembers = () => {
+  const api = usePubliqApi();
+  const { data: user } = useGetMe();
+  const [hasFamilyMembers, setHasFamilyMembers] = useState(false);
```
[the remark this answers is not in the document]
Comment (chennara) [reply src/onboarding/family/_queries/useHasFamilyMembers.ts:10]: I think so, `hasFamilyMembers` is set in the `onError` callback on line 14.

File: src/_context/OnboardingProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const OnboardingContext = createContext<TOnboardingContext>({
+  dismissFamilyOnboarding: () => {},
+  withFamilyOnboarding: !storage.getBoolean(StorageKey.HasSeenFamilyOnboarding),
+});
+
+export const useOnboarding = () => useContext(OnboardingContext);
+
+const OnboardingProvider: FC<PropsWithChildren> = ({ children }) => {
+  const [withFamilyOnboarding, setWithFamilyOnboarding] = useState(!storage.getBoolean(StorageKey.HasSeenFamilyOnboarding));
```
Comment (samvanhoeyicapps) [inline src/_context/OnboardingProvider.tsx:19]: maybe showFamilyOnboarding?

File: src/_context/OnboardingProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import { storage } from '../storage';
+
+type TOnboardingContext = {
+  dismissFamilyOnboarding: () => void;
+  withFamilyOnboarding: boolean;
+};
+
+const OnboardingContext = createContext<TOnboardingContext>({
+  dismissFamilyOnboarding: () => {},
+  withFamilyOnboarding: !storage.getBoolean(StorageKey.HasSeenFamilyOnboarding),
```
Comment (samvanhoeyicapps) [inline src/_context/OnboardingProvider.tsx:13]: withFamilyOnboarding gets set twice to true now, in initial state and here

File: src/onboarding/family/FamilyOnboarding.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+  const { dismissFamilyOnboarding } = useOnboarding();
+  const resolveFamilyOnboarding = useCallback(() => {
+    dismissFamilyOnboarding();
+    storage.set(StorageKey.HasSeenFamilyOnboarding, true);
+  }, [dismissFamilyOnboarding]);
+
+  useEffect(() => {
+    if (isFetchedAfterMount && hasFamilyMembers) {
+      resolveFamilyOnboarding;
```
Comment (samvanhoeyicapps) [inline src/onboarding/family/FamilyOnboarding.tsx:31]: also invoke the function

File: src/onboarding/family/FamilyOnboarding.tsx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+  const { t } = useTranslation();
+
+  if (isLoading) {
+    return <Spinner statusBarStyle={Platform.OS === 'ios' ? 'dark-content' : 'light-content'} />;
+  }
+
+  return (
+    <>
+      <Analytics screenName="FamilyOnboarding" />
```
Comment (samvanhoeyicapps) [inline src/onboarding/family/FamilyOnboarding.tsx:43]: are the screenNames publiq want defined?

**Review by samvanhoeyicapps — DISMISSED** [review]

**Review by samvanhoeyicapps — DISMISSED** [review]

**Review by samvanhoeyicapps — APPROVED** [review]

**Pull request merged**